torch/distributed/pipelining/PipelineStage.py

```diff
@@ -549,6 +576,42 @@ def backward_one_chunk(
         )
         self.bwd_chunk_id += 1
 
+    def _validate_fwd_input(self, args, kwargs):
+        """Raises a RuntimeError if shapes of input args/kwargs do not match the shapes configured for this stage."""
+
+        if self.is_first:
+            # TODO why is there a separate recv_info for each pipeline chunk?
+            expected_args = self.args_recv_info[self.fwd_chunk_id]
+        else:
+            expected_args = tuple()
+
+        if len(kwargs):
+            # TODO- need a mapping of kwarg to position in self.args_recv_info
+            # without it, we just validate shapes for args and ignore kwargs
+            logger.warning(
+                "Unable to validate input info for traced modules with kwargs."
+            )
```

Comment:
Don't warn. May scare users.

Reply:
yea, i am ok removing this. i think technically we have a hole that if users use kwargs they may face shape errors, but rather than warning maybe we should file a todo issue for following up on that